py/py_sparkling/ml/algos.py

```diff
@@ -28,7 +30,42 @@ def propagate_value_from_deprecated_property(kwargs, from_deprecated, to_replaci
         del kwargs[from_deprecated]
 
 
-class H2OGBM(H2OGBMParams, JavaEstimator, JavaH2OMLReadable, JavaMLWritable):
+class H2OAlgorithmCommons(JavaEstimator, JavaH2OMLReadable, JavaMLWritable):
+
+    def getBinaryModel(self):
+        model_id = py4j.java_gateway.get_field(self._java_obj.getBinaryModel(), "_key").toString()
+
+        model_json = h2o.api("GET /%d/Models/%s" % (3, model_id))["models"][0]
+        metrics_class, model_class = H2OEstimator._metrics_class(model_json)
+        m = model_class()
+        m._id = model_id
+        m._model_json = model_json
+        m._have_pojo = model_json.get('have_pojo', True)
+        m._have_mojo = model_json.get('have_mojo', True)
+        m._metrics_class = metrics_class
+        m._start_time = model_json.get('output', {}).get('start_time', None)
+        m._end_time = model_json.get('output', {}).get('end_time', None)
+        m._run_time = model_json.get('output', {}).get('run_time', None)
+    
+        if model_id is not None and model_json is not None and metrics_class is not None:
```

Comment:
Can we break this if statement into smaller private methods?

Reply:
Yup, after having second look on the code it does code does not seem completely right even though based on H2O. Will verify. Thank you!